### PR TITLE
Convert History Internal Service API Protos to use ProtoTime

### PR DESCRIPTION
**Pull request by shawnhathaway**
**What changed?**
- History Internal Service APIs proto times are moved to use ProtoTimes with the Gogo Time extension to materialize Golang time types.
- PersistenceBlobs additionally now use the Gogo Time Extension

**Why?**
- Effort to use correct ProtoTime types

**How did you test it?**
- Local Testing, Buildkite

**Potential risks**
- Incorrect Expiry/Time calculations

### CODE REVIEW

File: common/log/tag/tags.go [hunk 1]
```diff
@@ -58,6 +59,11 @@ func Timestamp(timestamp time.Time) Tag {
 	return newTimeTag("timestamp", timestamp)
 }
 
+// Timestamp returns tag for Timestamp
+func TimestampPtr(t *time.Time) Tag {
```
Comment (alexshtin) [inline common/log/tag/tags.go:63]:
```suggestion
func TimePtr(t *time.Time) Tag {
```

File: common/backoff/cron.go [hunk 2]
```diff
@@ -78,12 +78,11 @@ func GetBackoffForNextSchedule(cronSchedule string, scheduledTime time.Time, now
...
 
-// GetBackoffForNextScheduleInSeconds calculates the backoff time in seconds for the next run given
-// a cronSchedule, current scheduled time, and now.
-func GetBackoffForNextScheduleInSeconds(cronSchedule string, scheduledTime time.Time, now time.Time) int32 {
+// GetBackoffForNextScheduleNonNegative calculates the backoff time and ensures a non-negative duration.
+func GetBackoffForNextScheduleNonNegative(cronSchedule string, scheduledTime time.Time, now time.Time) *time.Duration {
 	backoffDuration := GetBackoffForNextSchedule(cronSchedule, scheduledTime, now)
-	if backoffDuration == NoBackoff {
-		return 0
+	if backoffDuration == NoBackoff || backoffDuration < 0 {
```
Comment (alexshtin) [inline common/backoff/cron.go:84]: Shouldn't happen, but ok.

File: common/persistence/sql/sqlExecutionManager.go [hunk 3]
```diff
@@ -1057,14 +1055,14 @@ func (m *sqlExecutionManager) GetTimerIndexTasks(
...
-		goVisibilityTimestamp, err := types.TimestampFromProto(resp.Timers[request.BatchSize].VisibilityTime)
+		goVisibilityTimestamp := resp.Timers[request.BatchSize].VisibilityTime
 		if err != nil {
 			return nil, serviceerror.NewInternal(fmt.Sprintf("GetTimerTasks: error converting time for page token: %v", err))
 		}
 
 		pageToken = &timerTaskPageToken{
 			TaskID:    resp.Timers[request.BatchSize].GetTaskId(),
-			Timestamp: goVisibilityTimestamp,
+			Timestamp: *goVisibilityTimestamp,
```
Comment (alexshtin) [inline common/persistence/sql/sqlExecutionManager.go:1065]: Safe?

File: service/history/timerQueueStandbyTaskExecutor.go [hunk 4]
```diff
@@ -132,9 +132,8 @@ func (t *timerQueueStandbyTaskExecutor) executeUserTimerTimeoutTask(
 				return nil, serviceerror.NewInternal(errString)
 			}
 
-			t, _ := types.TimestampFromProto(timerTask.VisibilityTime)
 			if isExpired := timerSequence.isExpired(
-				t,
+				*timerTask.VisibilityTime,
```
Comment (alexshtin) [inline service/history/timerQueueStandbyTaskExecutor.go:136]: Safe?

File: common/log/tag/tags.go (hunk 1 above)
Comment (shawnhathaway) [reply common/log/tag/tags.go:63]: Seems func naming is based off tag string key, in this case - `"timestamp"`

File: service/history/timerQueueStandbyTaskExecutor.go (hunk 4 above)
Comment (shawnhathaway) [reply service/history/timerQueueStandbyTaskExecutor.go:136]: Should be but added a TimeValue call

File: common/persistence/sql/sqlExecutionManager.go (hunk 3 above)
Comment (shawnhathaway) [reply common/persistence/sql/sqlExecutionManager.go:1065]: Should be but added a TimeValue call

**Pull request merged**